Source/Utils/String+Extension.swift

```diff
@@ -21,11 +21,14 @@
 import Foundation
 
 extension String {
-    var decodeString: String? {
-        return self.stringByReplacingOccurrencesOfString("+", withString: " ").stringByRemovingPercentEncoding
+	private static let allowedQueryCharactersPlusSpace = CharacterSet(charactersIn: "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789-._~ ")
```

Comment:
TL;DR: Apple is crazy and doing URL encoding correctly is hard

Okay, so there is a long history of Apple very badly mishandling how components of URLs are encoded. Turns out each portion of URLs have certain rules for encoding, and there are a lot of different components to a URL, and all of them have different rules. Apple treated them all the same, and BROKE a lot of the rules and were generally menaces to correctness, but worked 95% of the time. They tried to fix that, but frankly didn't fix it in their new stuff either. Add to that the fact that servers often screw that up as well, and you have the above. It works, it's aggressive on what it encodes (but that's perfectly valid) and it takes care of the weird stuff apple does since they don't give us all the tools we need to correctly encode query parameters. 

Note the old code used the HOST allowed characters for the query parameter, which is clearly wrong. Anyway, it would be a good subject for a long blog post or something, but this code works! :D